src/content/posts/whats-possible-with-melange.mdx

```diff
@@ -0,0 +1,202 @@
+---
+title: "What's possible with Melange"
+publishedAt: "2023-12-18"
+tags:
+  - "Melange"
+  - "dune"
+  - "OCaml"
+  - "Reason"
+---
+
+## Introduction
+
+[Melange](https://melange.re) is a backend for the OCaml compiler that emits readable, modular and highly performant JavaScript code. It’s integrated with all OCaml-related tooling such as opam to install packages and dune as a build system.
+
+This blog post is an overview of the capabilities of Melange and dune together, what's possible to do now and it wasn't with the previous solution (ReScript v9).
+
+This blog post isn't a direct comparison between Melange v2 and ReScript v9 (also know as BuckleScript) compilers, it wouldn’t be fair since ReScript v9 is not the latest version and their goals were not designed to solve any of the problems Melange solves.
+
+Melange started as a fork of BuckleScript but focusing in OCaml compatibility while ReScript continued the integration with JavaScript ecosystem.
+
+## Works with the latest versions of OCaml
+
+One blocker for some OCaml developers to use ReScript is the version of the OCaml compiler being stuck in 4.06, a few versions behind latest (5.1) OCaml.
+
+This was an issue in our backend as well, since some packages needed to be compatible for 4.06 and 5.1 and a lot has changed:
+
+- Improvements in type-checking (GADTs, local modules, type variables)
+- [Binding operators (let*, let+, and*, let.async, let.opt, etc…)](https://v2.ocaml.org/releases/4.11/htmlman/bindingops.html)
+- [A new quoted extension syntax](https://github.com/ocaml/ocaml/pull/8820)
+- [Support for macOS/arm64](https://github.com/ocaml/ocaml/pull/9699)
+- Improved error messages
+- [Many](https://ocaml.org/releases/4.09.1) [many](https://ocaml.org/releases/4.10.1) [bug](https://ocaml.org/releases/4.11.0) [fixes](https://ocaml.org/releases/4.12.0)
+- [Tail modulo cons (TMC) optimisation](https://v2.ocaml.org/manual/tail_mod_cons.html)
+
+Side note: A few features of those years of OCaml development don't apply to Melange. In fact, OCaml 5 introduced the long-awaited called `multicore`. Currently multicore isn’t a feature supported by Melange since it can’t benefit extensively, due to the nature of JavaScript being single threaded.
```

Comment:
```suggestion
Side note: A few features of those years of OCaml development don't apply to Melange. In fact, OCaml 5 introduced long-awaited `multicore` features. Currently multicore isn’t a feature supported by Melange since it can’t benefit extensively, due to the nature of JavaScript being single threaded.
```